wagtail/snippets/views/snippets.py

```diff
@@ -120,6 +121,19 @@ def _get_locale(self):
     def _get_is_searchable(self):
         return class_is_indexed(self.model)
 
+    def _get_search_form(self):
+        if self.is_searchable and "q" in self.request.GET:
+            return SearchForm(
```

Comment:
Unrelated to this PR, but does anyone think it's a bit weird that Wagtail gives you a "This field is required" error when you remove your search query?

To me, I think it would be normal for a user to remove their search query if they want to go back to the full listing. This works, but they would get this validation error.

I think if we make that search field non-required, this ``if`` statement wouldn't be needed.

Reply:
Yeah. I have also thought that.  Maybe it should be pulled out to a new issue though as it probably does the same thing for any header search.